### ISSUE TITLE
Allow additional arguments for external application

### ISSUE THREAD

**Issue by miceno** [collectiveaccess/providence#382]
While running application for the `external_applications.conf` file, I found that my shared hosting platform needed to limit the resources of some of the commands.

I propose a new feature to allow configuring additional parameters on `external_applications_conf`, that will allow additional parameters for commands like ImageMagick `convert`, `identify`, and the like.

The configuration will be something like for commands `convert` and `identify` on ImageMagick:

```
imagemagick_convert_args=-limit thread 1
imagemagick_identify_args=-limit thread 1
```

What do you think?

**Comment by collectiveaccess** [comment]
I agree we should add these. We also need to change the format so more than one path can be specified, and each can be tried in turn. This way we can ship with all common paths specified and save the hassle of having to tweak it. For example: the Ghostscript gs executable may be in /usr/bin or /usr/local/bin on Linux depending upon how it was installed (Package vs. source)

**Comment by miceno** [comment]
Ok, I am preparing a new branch with the args feature.

Concerning the paths, I think it will be trivial to allow a list of comma-separated paths too. 

I have been reviewing the Media Plugin code and I think it requires a complete refactor with regard to file checks, to make sure checks are run only once, and then cached, in a static variable, so you can use it on every next request to checkStatus. I will open another issue about it, just to avoid mixing conversations here.

But I will finish this first, as of now it is prioritary to me.

**Comment by collectiveaccess** [comment]
Supporting multiple paths is probably best done using a configuration list. Eg. what we have now:

ghostscript_app = /usr/bin/gs

becomes

ghostscript_app = [/usr/bin/gs, /usr/local/bin/gs]

It would not be hard to adapt current code to accept this, and to support older non-list files.

We regards to refactoring the media plugins, that code was written back in PHP 4.x times and updated piecemeal over time, so we are certainly all for working on it. Let's thoroughly discuss how to refactor before any work is started however.

**Comment by miceno** [comment]
Sure!

I will open a new issue for the refactor and discuss it there #383
